### ISSUE TITLE
What about extend functions?

### ISSUE THREAD

**Issue by Z128A** [markcwm/openb3d.mod#26]
Do I understand correctly that your library is written in blitz basic? Why not in C++?

What about add next functionallity:

Add random generator.
Read\Write files.
Date\Time functions.
Direct access (pointer) to screen buffer.

It may be all wrong, of course.
Although it probably doesn't make sense.
What do you think about it?

**Comment by Kippykip** [comment]
Because technically this project is a bmx wrapper to another project called openb3dlib, which was already cpp.

It just seems this wrapper was way more maintained and has quite a few bug fixes to the library itself in the end. Heh

**Comment by Z128A** [comment]
If I understand, that is:
https://github.com/markcwm/openb3d.mod/tree/master/openb3dlib.mod/openb3d/src

Thanks. You have fun avatar)

**Comment by Z128A** [comment]
I looked at this library and I was wrong, there are functions for working with files, as well and memory banks supports too. There are even shaders. But there is nothing to work with sound / music. By the way, if scan the dll for functions, then there are just a "sea" of functions. Especially a lot with regards to GL operations.

**Comment by Kippykip** [comment]
Yes sound and music isn't handled by OpenB3D. For myself I use fmod modules for blitzmax so I get DSP and added wall checks to muffle or dampen it. Using a bit of math from the cameras position I pan it as well.

Normally though BMX has it's own 2D sound functions based on freeaudio, but it doesn't seem to support streaming or DSP (at least the BMX module).

**Comment by Blitz3DFASM** [comment]
Now thats is clear. Why no one took the source code for embedding fmod to dll from the blitz3d open sources?

I see file functions in sources: https://github.com/markcwm/openb3d.mod/blob/master/openb3dlib.mod/openb3d/src/file.cpp but i not see it inside DLL... Also nothing about Text output, fonts, keyboard mouse input etc...

**Comment by Kippykip** [comment]
Because all those functions are already built into blitzmax or can be used natively with cpp projects?
OPENB3D is just for the 3D stuff
